janetechinc/roots-cognito-lambda-triggers
# feat(cognito): verify SSO users and migrate

Following this https://app.clickup.com/t/86aya49x7

### Summary:
We want to support partner's using SSO, for now just google.
The current migration lambda does not run when a user signs in via google sso, as it will always go to pre-signup.
We can handle this issue by allowing an SSO user to be created in pre-sign up, then we can migrate the user data in the post confirmation lambda.

### Changes:

Creates an interface to calm TS down with the api call (UserResponse)
Create new API call `get_sso_user_attributes` to retrieve user attributes for migration
Adds attributes to event being returned to Cognito
Update user on the partner's Cognito pool (this will require the partner to ensure their lambda role has the `cognito-idp:AdminUpdateUserAttributes` permission)
Updated the test


### How to Test
This branch is currently pushed to staging, except with mock data instead of hitting the Jane backend (waiting for endpoint to be deployed).
The cognito pool jane-sandbox on AWS is linked to it and has its lambdas working as expected now

- Create an account using google SSO on https://staging.nonprod-iheartjane.com/embed/stores/68
- Go on AWS -> Cognito -> jane-headless-test-staging
- Run the [headless-checkout-with-cognito-public](https://github.com/janetechinc/headless-checkout-with-cognito-public) repo locally, with the config pointing to the headless pool listed above ^
- Sign in via sso with same email as above
- Back on Cognito user pool, see the user was created with the proper fields
- On active admin, see a new user was created with the proper fields
